internal/servicename_test.go

```diff
@@ -36,6 +36,7 @@ func TestValidServiceNames(t *testing.T) {
 		"a77ab7g4-51cb-4808-a9ef-875568bde54a", // not valid UUID
 		"eviluuid26695g10-a384-48e7-8867-6d48b7fae80a", // not valid UUID
 		"a77gb7e4-51cb-4808-a9ef-875568bde54aeviluuid", // not valid UUID
+		"apb@uber.com",
```

Comment:
we should add some more obtuse emails. if emails are the desired test case, there are some really crazy chars allowed in email addrs, we should enumerate more explicit list here and in invalid based on our case.

Reply:
Did you have a sample test cases that this may not cover? I was striving for simplicity here so I wouldn't need to branch and write an email address parser/regex -- also supporting more unusual characters _could_ cause collisions in metrics, eg `alex+@uber.com` and `alex$@uber.com` would both show up as `alex__uber.com` in metrics

Reply:
We do have the option of using https://golang.org/pkg/net/mail/ to validate any email address. But, @AlexAPB has a point that we don’t want just any valid email address.

Reply:
Would it be better to provide a function to "clean" a string to be a valid service name, and tell service owners who want to use emails etc to use that function.

We want to allow this for metrics, but it'll cause problems if service owners try to listen using a name like this, but it isn't supported by the rest of our Infra stack.